### PR TITLE
Database Schema

### PR DESCRIPTION
**Pull request by Snugug**
Add Database Schema creation, and refactor the _crap_ outta everything to support that

**Changelog**

```markdown
**New**

* Database Schema Creation on Initialization 

**Changed**

* Index now returns a promise. :boom:

**Removed**

* Remove old tests
* Remove old schema
```

---
Resolves: #19 

`DCO 1.1 Signed-off-by: Sam Richard <snugug@us.ibm.com>`

### CODE REVIEW

File: lib/database.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+const tables = require('./database/tables');
+const schemas = require('./database/schemas');
+const utils = require('./utils');
+
+const knex = require('knex');
+
+const database = knex(config.knex);
+
+const init = () => {
```
Comment (scottnath) [inline lib/database.js]: add jsdoc please

File: lib/database/schemas.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  {
+    name: 'created',
+    type: 'timestamp',
+  },
+  {
+    name: 'language',
+    type: 'string',
+  },
+  {
+    name: 'sunrise',
```
Comment (scottnath) [inline lib/database/schemas.js]: At one point you said `sunrise` would be `required`. Still so?

File: lib/database/schemas.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    name: 'id',
+    type: 'string',
+  },
+  {
+    name: 'created',
+    type: 'timestamp',
+  },
+  {
+    name: 'language',
+    type: 'string',
```
Comment (scottnath) [inline lib/database/schemas.js]: Language should be a `select` or point to a taxonomy or some such.
Does _not_ have to be for this pull request...but will need to happen

File: lib/database/schemas.js (hunk 2 above)
Comment (scottnath) [inline lib/database/schemas.js]: If `required` doesn't belong here...where does it belong?

File: lib/database/schemas.js (hunk 3 above)
Comment (Snugug) [inline lib/database/schemas.js]: This is Database schema, _not_ Content Model schema

File: lib/database/schemas.js (hunk 2 above)
Comment (Snugug) [inline lib/database/schemas.js]: This is database schema, _not_ Content Model schema

Comment (scottnath) [inline lib/database/schemas.js]: Cool....where will these common content model schemas live?

Comment (Snugug) [inline lib/database/schemas.js]: Probably inside `lib/content-types`. They aren't schema. The user mocking will live inside `lib/auth` or `lib/user` or something like that.

File: lib/database.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    // Loop over all schema and create database tables
+    return Promise.map(schemata, schema => {
+      return tables.create(database, schema);
+    });
+  }).then(() => {
+    // Get raw content types
+    return types.raw();
+  }).then(raw => {
+    // Store versioned content types in DB
+    return database('all-types').insert({
```
Comment (scottnath) [inline lib/database.js]: Can this please move this to initialization.

**Comment by scottnath** [comment]
Just those notes. Jsdocs missing throughout, but can be fixed. Only real item is moving `raw` out of database initialization.
:hammer:

**Comment by scottnath** [comment]
@Snugug :/ failing test

**Comment by Snugug** [comment]
I fixed the damn race condition

**Pull request merged**